netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -118,45 +128,87 @@ public void teardown() throws Exception {
     group.shutdownGracefully(0, 10, TimeUnit.SECONDS);
   }
 
+  @Test
+  public void headersShouldAddDefaultUserAgent() throws Exception {
+    NettyClientTransport transport = createTransport();
+    transport.start(clientTransportListener);
+
+    // Send a single RPC and wait for the response.
+    unaryCall(transport, new Metadata.Headers()).get(10, TimeUnit.SECONDS);
+
+    // Verify that the received headers contained the User-Agent.
+    assertEquals(1, serverListener.streamListeners.size());
+
+    Metadata.Headers headers = serverListener.streamListeners.get(0).headers;
+    assertEquals(Utils.DEFAULT_USER_AGENT.toString(UTF_8), headers.get(HttpUtil.USER_AGENT));
+  }
+
+  @Test
+  public void headersShouldOverrideDefaultUserAgent() throws Exception {
+    NettyClientTransport transport = createTransport();
+    transport.start(clientTransportListener);
+
+    // Send a single RPC and wait for the response.
+    String userAgent = "testUserAgent";
+    Metadata.Headers sentHeaders = new Metadata.Headers();
+    sentHeaders.put(HttpUtil.USER_AGENT, userAgent);
+    unaryCall(transport, sentHeaders).get(10, TimeUnit.SECONDS);
+
+    // Verify that the received headers contained the User-Agent.
+    assertEquals(1, serverListener.streamListeners.size());
+    Metadata.Key<String> userAgentKey = Metadata.Key.of(Utils.USER_AGENT.toString(UTF_8),
```

Comment:
I guess you don't to make sure the "user-agent" strings are the same? Do we want to use HttpUtil.USER_AGENT.name() some places?